source/common/http/http1/codec_impl.cc

```diff
@@ -187,7 +188,31 @@ void StreamEncoderImpl::encodeData(Buffer::Instance& data, bool end_stream) {
   }
 }
 
-void StreamEncoderImpl::encodeTrailers(const HeaderMap&) { endEncode(); }
+void StreamEncoderImpl::encodeTrailers(const HeaderMap& trailers) {
+  if (!connection_.enable_trailers_) {
+    return endEncode();
+  }
+  // Trailers only matter if it is a chunk transfer encoding
+  // https://tools.ietf.org/html/rfc7230#section-4.4
+  if (chunk_encoding_) {
```

Comment:
So concern about this which I think we can address with better docs.

Envoy doesn't do chunk encoding by default; any time there is a content length with the response, it will not chunk encode.  That means that if you're regularly sending trailers, some will make it through and others will not.

I think it's worth commenting somewhere (at config?) that trailers will only be sent where Envoy does chunked encoding, and an explanation of when Envoy does chunked encoding (where chunk_encoding != false in the HTTTP/1 codec)  Basically if someone wants to consistently propagate trailers, I think they're going to have to have a custom filter removing content length and it won't work for some corner cases like HTTP/1.0 and we should make that obvious.

Reply:
I wonder how much of this is Envoy specific or just a standard convention that can already be explain by looking at the Http specs. Either way, I'll add some comments in the `include/envoy/http/codec.h` Is this a good spot?